kafka/lazy_client.go

```diff
@@ -10,19 +10,26 @@ import (
 )
 
 type LazyClient struct {
-	once    sync.Once
+	mutex   sync.Mutex
 	initErr error
 	inner   *Client
 	Config  *Config
 }
 
 func (c *LazyClient) init() error {
-	var err error
+	func(c *LazyClient) {
```

Comment:
why declaring a function in this function to just call it?

Reply:
You are setting c.initErr in this function but using the value below. There is a possible race condition if the value is set from exiting this function to reading the value below.

Reply:
I think I would lock the mutex for the whole duration of the init() call.

Reply:
Good catch. I see how a race condition could happen and a transient error that was encountered might not be fully logged. Performance isn't critical here so I will change it.